os/drivers/wireless/scsc/t20_ops.c

```diff
@@ -1257,7 +1257,6 @@ int slsi_get_signal_poll(void *priv, struct wpa_signal_info *si)
 	if (!ndev_vif->activated) {
 		SLSI_ERR_NODEV("STA Not Connected\n");
 		return -EINVAL;
-		goto exit;
```

Comment:
No removal of this line is wrong, because `SLSI_MUTEX_UNLOCK(ndev_vif->vif_mutex);` should be called before return.
So, this should be changed as below,
```
res = -EINVAL;
goto exit;
```